sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -45,6 +45,78 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
     withOrigin(ctx)(StructType(visitColTypeList(ctx.colTypeList)))
   }
 
+  override def visitSingleRoutineParamList(ctx: SingleRoutineParamListContext): StructType = {
+    withOrigin(ctx)(StructType(visitSimplifiedColDefinitionList(ctx.colDefinitionList())))
+  }
+
+  private def visitSimplifiedColDefinitionList(
+      ctx: ColDefinitionListContext): Seq[StructField] = withOrigin(ctx) {
+    ctx.colDefinition().asScala.map(visitStructField).toSeq
+  }
+
+  /**
+   * Create a [[StructField]] from a column definition which allows options like COMMENT and
+   * DEFAULT.
+   *
+   * Don't handle generation expression since this function is currently only used for creating
```

Comment:
why don't we reject it here within this function?

Reply:
we reject it in SparkSqlParser so that the error ParserRuleContext is the same as in DBR (CreateUserDefinedFunctionContext.parameters)